### DISCUSSION TITLE
cmdVelocityworld function will cause one of drones to lose constant height

### DISCUSSION THREAD

**Discussion by wawachen** [USC-ACTLab/crazyswarm#557] [Q&A]
Hi,

I observed when I use more than one crazyflies. The cmdVelocityworld function will cause several drones to lose height, which means I always sent the z velocity to be zero, but it will fall a little bit and hover at a lower height. (For one drone, it was fine)

could you please help this?

**Comment by whoenig** [comment]
If you use the on-board velocity controller, you likely still need an outer position controller for the height. Alternatively, you can use cmdFullState and send position and velocity setpoints.
